### PR TITLE
editor: affilition suggester

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds a suggester for instititution which suggestet based on
institution names and institution_acronym and returns results with
payload that is populated with necessary fields for UI.

Signed-off-by: harunurhan <harunurhan17@gmail.com>

@jmartinm  please check if the output and input make sense for the case. And I am assuming `institution_acronym` is present all the time, is it OK?

### CODE REVIEW

**Comment by jmartinm** [comment]
The field that should end up in the form input is `legacy_ICN`.

For input put all the ones that have `"copy_to": "affautocomplete"` https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/mappings/records/institutions.json#L14 plus all the fields in the address (https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/mappings/records/institutions.json#L31)

**Comment by harunurhan** [comment]
@jmartinm  I think now it's good.

File: inspirehep/modules/records/mappings/records/institutions.json [hunk 1]
```diff
@@ -57,6 +57,12 @@
                 "affautocomplete": {
                     "type": "string"
                 },
+                "affiliation_suggest": {
+                    "analyzer": "simple",
```
Comment (jacquerie) [inline inspirehep/modules/records/mappings/records/institutions.json]: This is like this by default, there's no need to specify it explicitly.

File: inspirehep/modules/records/mappings/records/institutions.json [hunk 2]
```diff
@@ -57,6 +57,12 @@
...
+                    "payloads": true,
+                    "search_analyzer": "simple",
```
Comment (jacquerie) [inline inspirehep/modules/records/mappings/records/institutions.json]: Same as above.

File: tests/unit/records/test_records_receivers.py [hunk 3]
```diff
@@ -541,7 +542,8 @@ def test_populate_experiment_suggest_populates_if_record_is_experiment():
     assert json_dict['experiment_suggest']['input'] == \
         ['foo', 'bar', 'foo_var', 'bar_var']
     assert json_dict['experiment_suggest']['output'] == 'foo'
-    assert json_dict['experiment_suggest']['payload']['$ref'] == 'http://foo/$ref'
+    assert json_dict['experiment_suggest'][
+        'payload']['$ref'] == 'http://foo/$ref'
```
Comment (jacquerie) [inline tests/unit/records/test_records_receivers.py]: Unrelated.

File: tests/unit/records/test_records_receivers.py [hunk 4]
```diff
@@ -36,6 +36,7 @@
     references_validator,
     populate_experiment_suggest,
     populate_abstract_source_suggest,
+    populate_affiliation_suggest
```
Comment (jacquerie) [inline tests/unit/records/test_records_receivers.py]: Nit: missing comma.

File: tests/unit/records/test_records_receivers.py [hunk 5]
```diff
@@ -598,3 +600,111 @@ def test_populate_abstract_source_suggest_does_nothing_if_record_is_not_hep():
         {'source': 'foo'},
         {'source': 'bar'},
     ]
+
+
+def test_populate_affiliation_suggest():
```
Comment (jacquerie) [inline tests/unit/records/test_records_receivers.py]: This test is **really** hard to follow. I will split it in several, each dealing with a different field. This is not just for readability: I will also add tests to ensure that the schema is what is asserted by the code, because the `institutions` schema is going to change soon, so it would be nice to have good errors on this.

File: inspirehep/modules/records/receivers.py [hunk 6]
```diff
@@ -267,6 +268,55 @@ def populate_abstract_source_suggest(sender, json, *args, **kwargs):
...
+            'affiliation_suggest': {
+                'input': inputs,
+                'output': legacy_ICN,
+                'payload': {
+                    'institution': institution,
+                    'institution_acronym': institution_acronym,
+                    'ICN': ICN,
+                    'legacy_ICN': legacy_ICN,
+                    'department': json.get('department'),
+                    '$ref': get_value(json, 'self.$ref')
```
Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: Nit: missing comma.

File: tests/unit/records/test_records_receivers.py [hunk 7]
```diff
@@ -598,3 +600,111 @@ def test_populate_abstract_source_suggest_does_nothing_if_record_is_not_hep():
...
+                '$ref': 'http://foo/bar',
+            },
+        }
+    }
+
+
+def test_populate_affiliation_suggest_does_nothing_if_record_is_not_institution():
+    json_dict = {
+        '$schema': 'http://localhost:5000/schemas/records/other.json',
+        'other': 'foo'
```
Comment (jacquerie) [inline tests/unit/records/test_records_receivers.py]: We don't even need this key, no?

Comment (harunurhan) [inline tests/unit/records/test_records_receivers.py]: not really no.

File: tests/unit/records/test_records_receivers.py (hunk 3 above)
Comment (harunurhan) [inline tests/unit/records/test_records_receivers.py]: do we have formatting script or something? to run before because usually when this always happen to me when I format with my text editor.

File: tests/unit/records/test_records_receivers.py (hunk 5 above)
Comment (harunurhan) [inline tests/unit/records/test_records_receivers.py]: I totally agree with splitting 👍  it wasn't even easy to write and change it for me
